### ISSUE TITLE
Enable configuration of mTLS support

### ISSUE THREAD

**Issue by csterwa** [cloudfoundry-incubator/haproxy-boshrelease#162]
`X-Forwarded-Client-Cert` headers are not injected/forwarded by HAProxy when a mTLS connection is established. HAProxy should behave as goRouter does and inject the client cert into an `X-Forwarded-Client-Cert` header given it has been configured to do so in the BOSH manifest.

**Comment by domdom82** [comment]
Not true. HAproxy will forward the client cert in DER binary format.
However, since HAproxy usually terminates TLS, the client cert CA needs to be configured in the crt_list or client_ca_file property. See https://github.com/cloudfoundry-incubator/haproxy-boshrelease/blob/master/jobs/haproxy/spec#L97

HAproxy will validate the client cert against the client CA and if the cert is valid, it will be forwarded in the `X-Forwarded-Client-Cert` header

**Comment by geofffranks** [comment]
Is this related to #163?

**Comment by domdom82** [comment]
I don't think so. #163 is fixing an edge-case where the X-Forwarded-Client-Cert is really big. This issue complains about not being able to forward the cert at all (which is not the case).
@csterwa Have you tried to configure your HAproxy as described above? It should work.

**Comment by plowin** [comment]
Closing after 1 year without feedback.
